### PR TITLE
koord-manager: add controllers options

### PR DESCRIPTION
**Pull request by saintube**
### Ⅰ. Describe what this PR does

Move the controller registration of the koord-manager into the options pkg instead of the main.go.

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: cmd/koord-manager/extensions/extension.go [hunk 1]
```diff
@@ -19,8 +19,24 @@ package extensions
...
 	"k8s.io/client-go/rest"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/koordinator-sh/koordinator/pkg/slo-controller/nodemetric"
+	"github.com/koordinator-sh/koordinator/pkg/slo-controller/noderesource"
+	"github.com/koordinator-sh/koordinator/pkg/slo-controller/nodeslo"
+)
+
+var (
```
Comment (zwzhang0107) [inline cmd/koord-manager/extensions/extension.go:32]: options.go
extensions pkg will be deprecated in future

Comment (saintube) [reply cmd/koord-manager/extensions/extension.go]: done

File: cmd/koord-manager/options/options.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+func NewOptions() *Options {
+	return &Options{
+		ControllerAddFuncs: controllerAddFuncs,
+		Controllers:        sets.StringKeySet(controllerAddFuncs).List(),
+	}
+}
+
+func (o *Options) InitFlags(fs *flag.FlagSet) {
+	pflag.StringSliceVar(&o.Controllers, "controllers", o.Controllers, fmt.Sprintf("A list of controllers to enable. '*' enables all on-by-default controllers,"+
```
Comment (zwzhang0107) [inline cmd/koord-manager/options/options.go:43]: add some explanation of full examples: 
case1: -controllers=*,-noderesource, means enables all controllers except noderesource
case2: -controllers=noderesource, means only enables noderesource

Comment (saintube) [reply cmd/koord-manager/options/options.go]: done

**Review by hormes — APPROVED** [review]

**Review by jasonliu747 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**